internal/pkg/service/objectstorage/cloud/s3.go

```diff
@@ -0,0 +1,141 @@
+package cloud
+
+import (
+	"context"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/smithy-go"
+
+	"github.com/giantswarm/object-storage-operator/api/v1alpha1"
+)
+
+type S3ObjectStorageAdapter struct {
+	*s3.Client
+	CloudRegion string
```

Comment:
We may prefer to avoid exporting this fields publicly. Instead we could
```suggestion
	s3client *s3.Client
	cloudRegion string
```

Reply:
Done